### DISCUSSION TITLE
ipv6 resolution with dnscrypt-server on docker

### DISCUSSION THREAD

**Discussion by ooonea** [DNSCrypt/dnscrypt-proxy#1840] [Q&A] [answered]
Hi,
I created the kenshiro, suami, julia and rei servers using the docker image on DockerHub and following the guide on https://github.com/DNSCrypt/encrypted-dns-server. The configuration seems to happen correctly, but although the server is reachable on IPv4 and IPv6, it only resolves addresses in IPv4. I realized this with two tests. Entering on dnscrypt-proxy only my servers in toml file, I get this result from https://ipv6-test.com/ (screenshot attached) and this other on https://test-ipv6.com/index.html.en_US (screenshot attached)
![Schermata da 2021-08-21 17-35-53](https://user-images.githubusercontent.com/35407790/130327803-ee7af4b2-ba24-45be-ad7e-1c1cbc66e818.png)
![Schermata da 2021-08-21 17-58-53](https://user-images.githubusercontent.com/35407790/130327804-02354863-22d3-4b3d-9fd2-7126c745664d.png)
. The tests tell me that the DNS servers do not have access to IPv6. Is there any particular setting to make them resolve IPv6-only addresses as well?

**Comment by jedisct1** [comment]
I'm not too familiar with Docker, but apparently, IPv6 support must be explicitly configured in the `docker` daemon: https://docs.docker.com/config/daemon/ipv6/

**Comment by ooonea** [comment, marked as answer]
Thank you very much @jedisct1 for your usual helpfulness (it's what makes dnscrypt a better protocol).
Your suggestion was very helpful, even if the answer is not on the official Docker guide. For anyone interested here I found my solution: https://medium.com/@skleeschulte/how-to-enable-ipv6-for-docker-containers-on-ubuntu-18-04-c68394a219a2
Now the sever are working as expected.

**Comment by ooonea** [comment]
I'd like to mark your answer, but GitHub decided to prevent me with a pop-up, which won't go away.
![immagine](https://user-images.githubusercontent.com/35407790/130331634-fa712ff2-c549-48f3-9a06-ba6dc477d430.png)
